javascript/javascript2/week3/warmup.js

```diff
@@ -0,0 +1,143 @@
+// Now create a new third function that has one parameter: planetLogFunction.
+// The only thing the third function should do is call the provided parameter function.
+// Try call the third function once with the Earth function and once with the Saturn function.
+
+const earthLogger = function() {
+    console.log('Earth');
+};
+const saturnLogger = function() {
+    console.log('Saturn');
+};
+
+const planetLogFunction = function(planetLogger) {
+    planetLogger();
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+// 5.Create a button with the text called "Log location".
+// When this button is clicked the location (latitude, longitude) of the user should be logged out using this browser api
+let map;
+const locationElement = document.querySelector('#location');
+function success(pos) {
+    const crd = pos.coords;
+    const center = {
+        lat: crd.latitude,
+        lng: crd.longitude
+    };
+    map.setCenter(center);
+    locationElement.innerHTML = `<p>This is the latitude ${crd.latitude}</p>`;
+    locationElement.innerHTML += `<p>This is the longitude ${crd.longitude}</p>`;
+    console.log('Your current position is:');
+    console.log(`Latitude : ${crd.latitude}`);
+    console.log(`Longitude: ${crd.longitude}`);
+    console.log(`More or less ${crd.accuracy} meters.`);
+}
+
+function error(err) {
+    locationElement.innerHTML = `ERROR(${err.code}): ${err.message}`;
+    console.warn(`ERROR(${err.code}): ${err.message}`);
+}
+
+const logLocationButton = document.querySelector('#log_button');
+logLocationButton.addEventListener('click', () => {
+    navigator.geolocation.getCurrentPosition(success, error);
+});
+
+// 6.Optional Now show that location on a map using fx the Google maps api
+function initMap() {
+    map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: 0, lng: 0 },
+        zoom: 20,
+    });
+}
+// 7.Create a function called runAfterDelay. It has two parameters: delay and callback.
+// When called the function should wait delay seconds and then call the provided callback function.
+// Try and call this function with different delays and different callback functions
+
+function runAfterDelay(delay, callback) {
+    setTimeout(callback, delay * 1000);
+}
+const logOut = function() {
+    console.log('Hi, I am here');
+}
+const countToTen = function() {
+    for (let i = 1; i <= 10; i++) {
+        console.log(i);
+    }
+}
+const soundYoda = function(str) {
+    const reversed = str.split(' ').reverse();
+    const yoda = reversed.join(' ');
+    console.log(yoda);
+}
+runAfterDelay(6, logOut);
+runAfterDelay(7, countToTen);
+runAfterDelay(8, () => soundYoda('I am your father'));
+
+// 8.Check if we have double clicked on the page. A double click is defined by two clicks within 0.5 seconds.
+// If a double click has been detected, log out the text: "double click!"
+
+window.addEventListener('dblclick', () => {
+    console.log('double click!')
+})
+
```

Comment:
In this exercise i think u also need to set time limit ...if two clicks within 0.5 seconds then its log out "double click" or if not then ????